src/JavaCodeGeneration.jl

```diff
@@ -0,0 +1,61 @@
+module JavaCodeGeneration
+
+using JavaCall.CodeGeneration
+using JavaCall.Reflection
+using JavaCall.Utils
+
+structidfromtypeid(typeid::Symbol) = Symbol(typeid, "Impl")
+
+paramnamefromindex(i::Int64) = Symbol("param", i)
+
+paramexprfromtuple(x::Tuple{Int64, ClassDescriptor}) = :($(paramnamefromindex(x[1]))::$(x[2].juliatype)) 
+
+function generateconvertarg(x::Tuple{Int64, ClassDescriptor})
+    :(push!(args, convert_to_jni($(x[2].jnitype), $(paramnamefromindex(x[1])))))
+end
+
+function methodfromdescriptors(
+    receiverdescriptor::ClassDescriptor,
+    descriptor::MethodDescriptor
+)
+    paramtypes = map(paramexprfromtuple, enumerate(descriptor.paramtypes))
+    pushfirst!(paramtypes, :(receiver::$(receiverdescriptor.juliatype)))
+    signature = string(
+        descriptor.rettype.signature,
+        '(',
+        receiverdescriptor.signature,
+        map(x->x.signature, descriptor.paramtypes)...,
+        ')')
```

Comment:
- Method signatures don't contain the type of the receiver.
- The return type comes after the arguments in the signature.

Reply:
Fixed in the next commit.